spec/controllers/external_authors_controller_spec.rb

```diff
@@ -0,0 +1,118 @@
+# frozen_string_literal: true
+require 'spec_helper'
+
+describe ExternalAuthorsController do
+  include LoginMacros
+  include RedirectExpectationHelper
+  let(:user) { FactoryGirl.create(:user) }
+
+  describe "GET #get_external_author_from_invitation" do
+    it "needs a valid invitaiton" do
+      get :get_external_author_from_invitation, invitation_token: "None existent"
+      it_redirects_to_with_error(root_path, "You need an invitation to do that.")
+    end
+   
+    it "needs to have a story assocated with the invitation" do
+      invitation = FactoryGirl.create(:invitation)
+      get :get_external_author_from_invitation, invitation_token: invitation.token
+      it_redirects_to_with_error(signup_path(invitation.token), "There are no stories to claim on this invitation. Did you want to sign up instead?")
+    end
+  end
+
+  describe "GET #complete_claim" do 
+    it "should give the user the works" do
+      external_author = FactoryGirl.create(:external_author)
+      user = FactoryGirl.create(:user)
+      fake_login_known_user(user)
+      invitation = FactoryGirl.create(:invitation, external_author: external_author)
+      get :complete_claim, invitation_token: invitation.token
+      it_redirects_to_with_notice(user_external_authors_path(user), "We have added the stories imported under #{external_author.email} to your account.")
+    end
+  end 
+
+  describe "GET #update" do
+    it "needs to be done by the right user" do
+      external_author = FactoryGirl.create(:external_author)
+      wrong_external_author = FactoryGirl.create(:external_author)
+      user = FactoryGirl.create(:user)
+      fake_login_known_user(user)
+      invitation = FactoryGirl.create(:invitation, external_author: wrong_external_author)
+      get :update, invitation_token: invitation.token, id: external_author.id
+      it_redirects_to_with_error(root_path, "You don't have permission to do that.")
+    end
+
+    it "can do nothing" do
```

Comment:
Above, we have 'context "user is not right"`, so down here we should have all of these inside the opposite context -- `context "user is right"`. (We need something more descriptive than "right" here, too, though.)

Also, this isn't actually that it does nothing or orphans the works. It's only testing the redirect and success message. So for the first two, we're probably looking at something like:
```
context "when user is right"
  context "when user chooses to do nothing with imported works"
    it "redirects with a success message"
  context "when user chooses to orphan imported works"
    it "redirects with a success message"
```
But `it "errors if the preferences can't be saved" do` might throw a bit of a wrench into that. I'll take a more thorough look at that later, but I wanted to get you started with a review.